packages/taquito/src/prepare/prepare-provider.ts

```diff
@@ -101,6 +106,49 @@ export class PrepareProvider implements PreparationProvider {
     return this.context.readProvider.getCounter(pkh, 'head') ?? '0';
   }
 
+  private adjustGasForBatchOperation(
+    gasLimitBlock: BigNumber,
+    gaslimitOp: BigNumber,
+    numberOfOps: number
+  ) {
+    return BigNumber.min(gaslimitOp, gasLimitBlock.div(numberOfOps + 1));
```

Comment:
I'm still worried that this kind of division can cause some problems. But this existed in previous code, so no action needed at this time.